index.html

```diff
@@ -1761,11 +1741,31 @@
             "PresentationRequest" data-lt="start">select a presentation
             display</a>, the <a>user agent</a> MUST <dfn>monitor the list of
             available presentation displays</dfn> by running the following
-            steps.
+            steps:
           </p>
           <ol link-for="PresentationAvailability">
-            <li>Set the <a>list of available presentation displays</a> to the
-            empty list.
+            <li>Let <var>availabilitySet</var> be a shallow copy of the <a>set
+            of presentation availability objects</a>.
+            </li>
+            <li>If there is a pending request to <a for="PresentationRequest"
+            data-lt="start">select a presentation display</a> for a
+            <a>PresentationRequest</a> and if the <a>PresentationRequest</a>'s
+            <a>presentation display availability</a> is <code>null</code>, then
+            run the following substeps:
+              <ol>
+                <li>Let <var>A</var> be a newly created
```

Comment:
Do we need a set though? There can be only one "pending presentation start request" per controlling browsing context (enforced by step 4 in the "select a presentation display" algorithm).

Or does that mean that I'm incorrectly restricting the monitoring algorithm and the "set of presentation availability objects" to the controlling browsing context?

No matter what, I agree we seem to have two algorithms into one: one "monitoring" algorithm that is meant to track availability changes of existing objects on a periodic basis, and one algorithm that we want to apply to a specific availability object (a temporary one when `start` is called, one from the set when `getAvailability` is called). It may be clearer to split things into two algorithms, although they obviously share the "Retrieve presentation displays" step.

(By the way, shouldn't the above mentioned step 4 be extended to reject the promise if there is an unsettled promise in nested contexts or in ancestor contexts as well?)